Library/Formula/poco.rb

```diff
@@ -31,5 +31,21 @@ def install
                           "--no-samples",
                           "--no-tests"
     system "make", "install", "CC=#{ENV.cc}", "CXX=#{ENV.cxx}"
+
+    #Some libraries remain linked against libraries in the buildpath.
+    #This loop changes the linking to /usr/local/lib
+    #https://github.com/Homebrew/homebrew/issues/35702
+    Dir["#{opt_prefix}/lib/*.dylib"].each do |poco_library|
+      dependencies = `otool -L #{poco_library}`
+      dependencies.split("\n").each do |dependency|
+        dependency = dependency.strip
+        if "#{buildpath}" == dependency[0,buildpath.to_s.length]
+          temporary_filename = dependency.strip.split(" ").first
+          install_filename = lib + File.basename(temporary_filename)
+          `install_name_tool -change #{temporary_filename} #{install_filename} #{poco_library}`
+        end
+      end
+    end
```

Comment:
Nice work here but it'd be great if we could submit this as an issue (and ideally a buildsystem patch) to the upstream developers so they can fix this properly. I'd hope this isn't something we'll have to do indefinitely. Thanks!